### PR TITLE
Updating migration docs to include most relevant breaking changes

### PR DESCRIPTION
**Pull request by tolzhabayev**
Adding documentation about most relevant breaking changes for plugin authors.

**Additional changes**
- Update the beta2 change log as the auto-generated parts of it does not seem to read well.

### CODE REVIEW

File: docs/sources/whatsnew/whats-new-in-v9-0.md [hunk 1]
```diff
@@ -215,6 +215,25 @@ In the InfluxDB data source, browser access mode was deprecated in grafana 8.0.0
 
 The rename by regex transformation has been improved to allow global patterns of the form `/<stringToReplace>/g`. Depending on the regex match used, this may cause some transformations to behave slightly differently. You can guarantee the same behavior as before by wrapping the match string in forward slashes (/), for example, (._) would become /(._)/. ([Github Issue #48179](https://github.com/grafana/grafana/pull/48179))
 
+### Clock Panel
+
+We have updated [clock panel](https://grafana.com/grafana/plugins/grafana-clock-panel/) to version `2.0.0` and encourage everyone to update as soon as possible. It will cause the Grafana to [crash](https://github.com/grafana/clock-panel/issues/106) when being used in a dashboard.
```
Comment (marefr) [inline docs/sources/whatsnew/whats-new-in-v9-0.md:220]: Get the impression that v2.0.0 will cause grafana to crash. Maybe something along the lines "Versions before 2.0.0 could crash on grafana v9 and above"

File: docs/sources/release-notes/release-notes-9-0-0-beta2.md [hunk 2]
```diff
@@ -83,28 +83,6 @@ module.exports.getWebpackConfig = (config, options) => ({
 
 Please refer to the webpack build error messages or the [official migration guide](https://webpack.js.org/migrate/5/) for assistance with fallbacks.
 
-**Which issue(s) this PR fixes**:
```
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply docs/sources/release-notes/release-notes-9-0-0-beta2.md:86]: Explain :)

File: docs/sources/whatsnew/whats-new-in-v9-0.md (hunk 1 above)
Comment (jackw) [reply docs/sources/whatsnew/whats-new-in-v9-0.md:220]: I agree. Right now this text is a bit misleading as v2.0.0 won't work in Grafana 7.

Comment (tolzhabayev) [reply docs/sources/whatsnew/whats-new-in-v9-0.md]: @marefr @jackw I have changed the text a bit, please read through it again https://github.com/grafana/grafana/pull/50356/commits/0339020b6b7c9377d7652343df8ba0f3e53eb6c3

File: docs/sources/release-notes/release-notes-9-0-0-beta2.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tolzhabayev) [reply docs/sources/release-notes/release-notes-9-0-0-beta2.md]: ok @marefr - I have reverted the manual change https://github.com/grafana/grafana/pull/50356/commits/5fbb05e9a6d7ceb28fc7a8850fb34dd13a64b1e2 and will run the Update changelog action separately.

[the remark this answers is not in the document]
Comment (tolzhabayev) [reply docs/sources/release-notes/release-notes-9-0-0-beta2.md]: https://github.com/grafana/grafana/pull/50398

**Review by jackw — APPROVED** [review]
Awesome stuff! 🚀

**Pull request merged**